SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/Operations/DataBrokerJobRunner.swift

```diff
@@ -37,23 +37,23 @@ public protocol WebJobRunner {
                 shouldRunNextStep: @escaping () -> Bool) async throws
 }
 
-@MainActor
 public final class DataBrokerJobRunner: WebJobRunner {
     let privacyConfigManager: PrivacyConfigurationManaging
     let contentScopeProperties: ContentScopeProperties
     let emailService: EmailServiceProtocol
     let captchaService: CaptchaServiceProtocol
 
-    internal init(privacyConfigManager: PrivacyConfigurationManaging,
-                  contentScopeProperties: ContentScopeProperties,
-                  emailService: EmailServiceProtocol,
-                  captchaService: CaptchaServiceProtocol) {
+    public init(privacyConfigManager: PrivacyConfigurationManaging,
+                contentScopeProperties: ContentScopeProperties,
+                emailService: EmailServiceProtocol,
+                captchaService: CaptchaServiceProtocol) {
         self.privacyConfigManager = privacyConfigManager
         self.contentScopeProperties = contentScopeProperties
         self.emailService = emailService
         self.captchaService = captchaService
     }
 
+    @MainActor
```

Comment:
I'm now curious what parts exactly we perform on the main thread versus not. I.e. to what extent are we meaningfully doing operations concurrently?
I guess it's probably fine since the bottleneck isn't really local processing

Reply:
Yeah exactly, in this case we're good since we're delegating this out to the concurrent thread pool via the `await` – this change here was simply pushing the MainActor requirement down from the entire class to the functions only, since it was otherwise impacting the `init` for this class. We don't need to `init` on the MainActor, we just want to keep that requirement for these functions.